### ISSUE TITLE
Environment initialization fails with custom path to pipenv

### ISSUE THREAD

**Issue by khinsen** [ObjectProfile/PythonBridge#8]
I have lots of Python installations and environments on my system, and the one I want to use for PythonBridge is not on $PATH for a number of good reasons. So I use `PBPharoPipenvProcess pipenvPath: '/PATH/TO/PIPENV/BINARY'` as instructed in the manual. But the initialization of that environment fails.

The cause lies in `PBPharoPlatform >> #forceInstallEnvironmentForApp:`. It runs `(self folderForApplication: application) / 'install_env.sh') fullName`, which in turn calls `pipenv` without an explicit path. Bash thus looks it up on $PATH, and that fails.

The best fix, in my opinion, would be to patch `install_env.sh` inside `forceInstallEnvironmentForApp:`, inserting the full path to `pipenv`. But perhaps this has some disadvantage that I don't see yet. If you think that's a good idea, I'd be happy to submit a PR.

Another fix would be to add the directory containing `pipenv` to the path of the shell process. The problem with that is that you never know which other executables are in that directory, so this could break something that `pipenv` relies on.

**Comment by khinsen** [comment]
I changed my mind: patching `install_env.sh` doesn't do the job if there's another `python` somewhere on the shell's `$PATH`. That leaves extending `$PATH` before running `install_env.sh`. That actually works fine. You can find the patch [here](https://github.com/feenkcom/PythonBridge/pull/1/commits/91bc338599639f5ddbe6a4f9e1d83e2b81c9be93).

**Comment by akevalion** [comment]
Hi @khinsen I have running PythonBridge in VisualWorks 8.x
![image](https://user-images.githubusercontent.com/10532890/106199091-b1712080-6193-11eb-8225-95ce1dd428d3.png)

I used to manipulate a picture from my computer

You can add your the path:

- Using the python installer to set the path. Reinstalling python can solve your problem, after the installation, program can set the variable
- Using the path of your system.

Here an example of my python path
![image](https://user-images.githubusercontent.com/10532890/106199570-67d50580-6194-11eb-9716-7a060f0e26f7.png)

**Comment by khinsen** [comment]
Thanks @akevalion! What you describe looks fine for Windows users. I am on macOS, where setting the path for GUI apps like Pharo is a royal pain, with a procedure that differs from one macOS version to another.

**Comment by akevalion** [comment]
I have followed the installation guide:
![image](https://user-images.githubusercontent.com/10532890/106554476-3c3a7e00-64fa-11eb-9092-b7e9241ecbeb.png)

Then I have opened the configuration file:`Pipfile` with my code editor, you can find this file in folder `PythonBridge`,  from there you can configure the python version, the modules, etc. Then you need to build the python project.

Run in your shell 
```
./install_env.sh 
```
Then you can get

![image](https://user-images.githubusercontent.com/10532890/106555566-950b1600-64fc-11eb-903b-ca833fe4d2f7.png)

After that there is another example with keras with another Pipfile.

Let me know if this helped you
https://objectprofile.github.io/PythonBridge/pages/pharo-installation#troubleshooting

**Comment by khinsen** [comment]
Thanks @akevalion ! I suppose that procedure would work for me as well - if I could put the Python installation I want to use on my shell's search PATH.

Unfortunately, I have at least five Python installations on my computer, because I often need to test Python code with different versions. The one that is on my PATH is installed via [Nix](https://nixos.org/), and is therefore not compatible with `pipenv`, so I cannot use it with `PythonBridge`. The four other ones are installed via [conda](https://docs.conda.io/en/latest/), which may or may not be compatible with `pipenv`; I am not really in the mood to try (and risk breaking installations I depend on).

Therefore I installed a fresh Python (via [MacPorts](https://www.macports.org/)), under `/opt/local/bin`, and configured `PythonBridge` to use it via `PBPharoPipenvProcess pipenvPath: '/opt/local/bin/pipenv'`. And that doesn't work because `install_env.sh` tries to use my Nix-installed Python instead of the MacPorts-installed Python.

**Comment by bergel** [comment]
Hi @khinsen ! 
How is it going? Do you still have the issue?

**Comment by khinsen** [comment]
No, because I use a patched version of `PythonBridge` that includes [this fix](https://github.com/feenkcom/PythonBridge/pull/1/commits/91bc338599639f5ddbe6a4f9e1d83e2b81c9be93). I haven't found another solution so far, and I am pretty sure that any solution in my situation requires some patch to `PythonBridge`. But there may well be a better fix than mine.

The fundamental issue is that `PythonBridge` uses different search methods in different places to locate the Python installation to be used. On every system where these search methods yield different outcomes, something will fail.
